File: Kattis/Egypt.hs
{-# LANGUAGE OverloadedStrings #-}

import           Control.Arrow         ((>>>))
import           Data.Bool             (bool)
import qualified Data.ByteString.Char8 as C
import           Data.Maybe            (fromJust)

main :: IO ()
main = C.getContents >>= (
                C.lines
            >>> filter (/= "0 0 0")
            >>> map (C.words
                >>> map readInt
                >>> (\[a,b,c] -> bool "wrong" "right" ((a^2 + b^2 == c^2) || (a^2 == b^2 + c^2) || (a^2 + c^2 == b^2))))
            >>> mapM_ C.putStrLn
        )

readInt :: C.ByteString -> Int
readInt = C.readInt >>> fromJust >>> fst

File: Competitions/Programming contest with Gnist 2/Egypt.py
def right(a,b,c):
    return a**2 + b**2 == c**2    

while True:
    a,b,c = (int(i) for i in input().split())
    if (a,b,c) == (0,0,0):
        break
    if right(a,b,c) or right(a,c,b) or right(b,a,c) or right(b,c,a) or right(c,a,b) or right(c,b,a):
        print("right")
    else:
        print("wrong")
